charts/azuremonitor-containers/templates/omsagent-secret.yaml

```diff
@@ -13,19 +13,14 @@ data:
   WSID: {{ required "A valid workspace id is required!" .Values.omsagent.secret.wsid | b64enc | quote }}
   KEY: {{ required "A valid workspace key is required!" .Values.omsagent.secret.key | b64enc | quote }}
   DOMAIN: {{ .Values.omsagent.domain | b64enc | quote }}
-  {{- $httpsProxyDict := urlParse .Values.Azure.proxySettings.httpsProxy  -}}
-  {{- $httpProxyDict := urlParse .Values.Azure.proxySettings.httpProxy  -}}
-  {{- if and (and (.Values.Azure.proxySettings.isProxyEnabled) (.Values.Azure.proxySettings.httpsProxy)) ($httpsProxyDict.userinfo) }} 
+  {{- if and (.Values.Azure.proxySettings.isProxyEnabled) (.Values.Azure.proxySettings.httpsProxy) }}
   PROXY: {{ .Values.Azure.proxySettings.httpsProxy | b64enc | quote }}
-  {{- else if and (and (.Values.Azure.proxySettings.isProxyEnabled) (.Values.Azure.proxySettings.httpsProxy)) (empty $httpsProxyDict.userinfo) }}   
-  # adding arbitrary creds since omsagent expects arbitrary creds in case of no auth
-  PROXY: {{ urlJoin (dict "scheme" $httpsProxyDict.scheme "userinfo" "admin:secret"  "host"  $httpsProxyDict.host) | b64enc | quote }}
-  {{- else if and (and (.Values.Azure.proxySettings.isProxyEnabled) (.Values.Azure.proxySettings.httpProxy)) ($httpProxyDict.userinfo) }}
+  {{- else if and (.Values.Azure.proxySettings.isProxyEnabled) (.Values.Azure.proxySettings.httpProxy) }}
   PROXY: {{ .Values.Azure.proxySettings.httpProxy | b64enc | quote }}
-  {{- else if and (and (.Values.Azure.proxySettings.isProxyEnabled) (.Values.Azure.proxySettings.httpProxy)) (empty $httpProxyDict.userinfo) }} 
-  # adding arbitrary creds since omsagent expects arbitrary creds in case of no auth
-  PROXY: {{ urlJoin (dict "scheme" $httpProxyDict.scheme "userinfo" "admin:secret"  "host"  $httpProxyDict.host) | b64enc | quote }}
   {{- else if ne .Values.omsagent.proxy "<your_proxy_config>" }}
   PROXY: {{ .Values.omsagent.proxy | b64enc | quote }}
   {{- end }}
+  {{- if and (.Values.Azure.proxySettings.isProxyEnabled) (.Values.Azure.proxySettings.proxyCert) }}
+  PROXYCERT.crt: {{.Values.Azure.proxySettings.proxyCert | b64enc | quote}}
```

Comment:
why would we ignore noproxy ?

Reply:
i see that we are ignoring. My question was why are we ignoring .

Reply:
> i see that we are ignoring. My question was why are we ignoring .
Because we dont use it. We only need HTTPS or HTTP proxy to use for outbound external endpoints.

Reply:
so if we want to scrape an ip address inside cluster, that should not go thru outbound proxy, how do we do it ?

Reply:
> so if we want to scrape an ip address inside cluster, that should not go thru outbound proxy, how do we do it ?

> Proxy configured for only external endpoints and everything else doesn't go through outbound proxy for example scrape IP address inside cluster.

Reply:
so with our current proxy approach, anything scraped outside the cluster will be broken, as there is no system wide proxy. so this will be broken anyways, and ssl is no first concern here.